### PR TITLE
FARM-88 Refactor/ Strings localisation

### PR DESCRIPTION
**Pull request by albertmiro**
[FARM-88] - Refactor task, support for languages.

#### 📲 What

Solve localisation issues and style format.

#### 🤔 Why
		
To make localisation work.
		
#### 🛠 How

We should use the Intl.message just with literals:
```	
class _LocalisedStrings{
 static String cancelString() => Intl.message('Cancel');
}
```
#### 👀 See
		
Screenshots / external resources
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/chat/bloc/handler/implementation/InteractiveMessageHandlerImpl.dart [hunk 1]
```diff
@@ -8,15 +8,22 @@ import 'package:farmsmart_flutter/chat/ui/widgets/text_input.dart';
 import 'package:intl/intl.dart';
 
 import '../InteractiveMessageHandler.dart';
+import 'package:intl/intl.dart';
+import 'package:intl/intl.dart';
```
Comment (albertmiro) [inline lib/chat/bloc/handler/implementation/InteractiveMessageHandlerImpl.dart]: Remove this duplicated import

File: lib/ui/profitloss/RecordTransactionHeader.dart [hunk 2]
```diff
@@ -2,14 +2,18 @@ import 'package:farmsmart_flutter/ui/profitloss/RecordTransaction.dart';
...
 import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
 
 class _Constants {
   static final amountValidator = RegExInputFormatter.withRegex(
       '^\$|^(0|([1-9][0-9]{0,3}))(\\.[0-9]{0,2})?\$');
 }
 
+class _LocalisedStrings {
+  static String hint() => Intl.message('00');
```
Comment (albertmiro) [inline lib/ui/profitloss/RecordTransactionHeader.dart]: I will move this one to _Strings

File: lib/chat/bloc/provider/ChatProvider.dart [hunk 3]
```diff
@@ -33,11 +33,15 @@ class _Constants {
   static const String typeValueMultiChoice = "com.wearemobilefirst.MultiChoice";
 }
 
-class _Strings {
-  static const String summaryTitleValue = "Chat summary value";
-  static const String summaryTitleLabel = "Chat summary";
-  static const String summaryActionButtonText = "Confirm";
-  static const String summaryError = "Provided summary is not correct";
+class _LocalisedStrings {
```
Comment (LeeMatthewHiggins) [inline lib/chat/bloc/provider/ChatProvider.dart]: this is not part of the farmsmart app. this is also not in any spec, its a screen Alex has made up for debugging. lets remove these as we don´t want them to translate debug.

File: lib/ui/LandingPage.dart [hunk 4]
```diff
@@ -7,11 +7,14 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:intl/intl.dart';
 
-class _Strings {
-  static final englishAction = "English";
-  static final swahiliAction = "Swahili";
-  static final confirmAction = "Confirm";
-  static final cancelAction = "Cancel";
+class _LocalisedStrings {
+  static String englishAction() => Intl.message('English');
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart]: probably don't want to translate the language names (as when switching you will only know your own language) eg. if you are English you might not know what English is, in Swahili in order to switch

File: lib/ui/mockData/MockUserProfileViewModel.dart [hunk 5]
```diff
@@ -68,27 +68,68 @@ class MockUserProfileListItemViewModel {
...
+  static String privacyPolicy() => Intl.message("Privacy Policy");
+
+  static String termsOfUse() => Intl.message("Terms of Use");
+
+  static String deleteProfile() => Intl.message("Delete Profile");
+
+  static String dummyTitle1() =>
+      Intl.message("Lorem ipsum dolor sit amet, consectetur adipiscing elit.");
+
+  static String dummyTitle2() => Intl.message(
```
Comment (LeeMatthewHiggins) [inline lib/ui/mockData/MockUserProfileViewModel.dart]: lets not translate the mock. this will put a lot of unneeded string in the translation list.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]

File: lib/ui/mockData/MockUserProfileViewModel.dart (hunk 5 above)
Comment (albertmiro) [reply lib/ui/mockData/MockUserProfileViewModel.dart]: I will remove as well the `Intl.message` in methods `buildLager` and `build` in this class:
- `Intl.message(_mockActionTitleLarger[index])`
- `Intl.message( _mockActionTitle[index])`

File: lib/chat/bloc/provider/ChatProvider.dart [hunk 6]
```diff
@@ -153,10 +149,9 @@ class ChatProvider implements ViewModelProvider<ChatViewModel> {
       child: Summary(
         viewModel: _chatSummaryProviderHelper.getSummary(
           inputModel: _responseMap,
-          titleValue: _LocalisedStrings.summaryTitleValue(),
-          titleText:
-              _LocalisedStrings.summaryTitleLabel().toUpperCase(),
-          actionText: _LocalisedStrings.summaryActionButtonText(),
+          titleValue: Intl.message(_Strings.summaryTitleValue),
+          titleText: Intl.message(_Strings.summaryTitleLabel.toUpperCase()),
+          actionText: Intl.message(_Strings.summaryActionButtonText),
```
Comment (albertmiro) [inline lib/chat/bloc/provider/ChatProvider.dart]: I will remove these ones too

File: lib/ui/LandingPage.dart [hunk 7]
```diff
@@ -159,7 +156,7 @@ class LandingPage extends StatelessWidget {
                       height: _Constants.subtitleLineSpace,
                     ),
                     Text(
-                      _viewModel.detailText,
+                      Intl.message(_viewModel.detailText),
```
Comment (albertmiro) [inline lib/ui/LandingPage.dart]: And these

**Review by LeeMatthewHiggins — APPROVED** [review]

**Review by davidfranquet — APPROVED** [review]
🚀

**Pull request merged**